src/Utility/TestMailer.php

```diff
@@ -94,6 +94,6 @@ protected function saveEmail($data)
             $state->emails = array();
         }
         $state->emails[] = array_filter($data);
-        $this->testSessionEnvironment->applyState($state);
+        $this->testSessionEnvironment->saveState($state);
```

Comment:
The change looks OK. The doc blocks aren't particularly useful and I'm not familiar with the difference, but presumably this has been tested in behat and fixes a problem, so LGTM